runsc/cmd/gofer.go

```diff
@@ -239,6 +240,7 @@ func (g *Gofer) Execute(_ context.Context, f *flag.FlagSet, args ...any) subcomm
 		UDSOpenEnabled:   conf.GetHostUDS().AllowOpen(),
 		UDSCreateEnabled: conf.GetHostUDS().AllowCreate(),
 		ProfileEnabled:   len(profileOpts) > 0,
+		PluginNetwork:    plugin.GetPluginStack() != nil,
```

Comment:
Thanks for pointing out. Yes, ideally gofer filter should not be affected by plugin stack. Here we add the seccomp rules are actually to make gofer workable after importing cgo package.
We found that as long as cgo package is imported, it will set iscgo as true in go runtime src/runtime/cgo.go. Later when newm1() is invoked, it will check whether iscgo is set. If it does set, newm1() will invoke an asmcgocall to start a cgo thread, which will invoke mmap syscall.
<img width="874" alt="截屏2024-01-05 下午7 40 43" src="https://github.com/google/gvisor/assets/130625925/29532162-c323-4fed-8120-1b7504b27356">
<img width="955" alt="截屏2024-01-05 下午5 47 54" src="https://github.com/google/gvisor/assets/130625925/e4781e66-ae4c-40a8-bbde-72c204cd9d5d">

However, since gofer process does not need to use cgo, we wonder whether there is a proper way to avoid setting iscgo as true in gofer.
One possible solution we came up with is to manually set iscgo as false during gofer initialization, but it's not very neat since M threads of gofer process created before manually setting iscgo will still start its corresponding cgo thread.

We are happy to hear any insightful comment or suggestion on removing cgo from gofer.

Reply:
Does it also invoke SYS_SET_ROBUST_LIST,  SYS_CLONE3, and SYS_RSEQ?

Anyways, this should be OK for now. I think we'd have to do some complicated stuff with the BUILD and maybe have 2 copies of runsc (with and without cgo) to avoid this.

And please add this info as a comment in `runsc/fsgofer/filter/config_cgo.go` so that future readers understand.

Reply:
Yes, it will also invoke these three syscalls. It will invoke clone3() which needs SYS_CLONE3 and clone3() will invoke start_thread(), which needs SYS_SET_ROBUST_LIST and SYS_RSEQ.
<img width="706" alt="截屏2024-06-06 下午3 01 55" src="https://github.com/google/gvisor/assets/130625925/040b4d97-846e-4430-ad92-5e0c650cb3d5">

<img width="564" alt="截屏2024-06-06 下午3 13 35" src="https://github.com/google/gvisor/assets/130625925/fa3638fa-9bf9-45da-afe3-5e6876f2cd40">

Thanks for advice. We will add this explanation in cgoFilters comment.

Reply:
Please see [this comment on `clone3`](https://github.com/google/gvisor/pull/9551#discussion_r1630495279); need a TODO here so that we can eventually remove clone3 from the allowed list.